mapper/software_management/src/software.rs

```diff
@@ -0,0 +1,93 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct SoftwareList {
+    pub modules: Vec<SoftwareModule>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct SoftwareModule {
+    pub software_type: String,
```

Comment:
you could also add new types for SoftwareType, SoftwareName, SoftwareVersion.... 

for instance `SoftwareOperation:SoftwareList { module_type }`. Should `module_type` be of same type as `SoftwareType` or is this something else?

Reply:
Good point. Will do.